content/code-signing-yaml/signing-macos.md

```diff
@@ -0,0 +1,193 @@
+    ```
+
+{{<notebox>}}
+Alternatively, each property can be specified in the `scripts` section of the YAML file as a command argument to programs with dedicated flags. See the details [here](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/app-store-connect/fetch-signing-files.md#--issuer-idissuer_id). In that case, the environment variables will be fallbacks for missing values in scripts.
+{{</notebox>}}
+
+## Publishing to Mac App Store
+
+In order to publish to Mac App Store, the application must be signed with a `Mac App Distribution` certificate using `Mac App Store` provisioning profile. Additionally, the application must be packaged into a `.pkg` Installer package which should be signed with `Mac Installer Distribution` certificate.
+
+### Fetch signing files automatically
+
+To code sign the app add the following commands in the [`scripts`](../getting-started/yaml#scripts) section of the configuration file, after all the dependencies are installed, right before the build commands.
+
+```yaml
+scripts:
+  ... your dependencies installation
+  - name: Set up keychain to be used for codesigning using Codemagic CLI 'keychain' command
+    script: keychain initialize
+  - name: Fetch Mac App Distribution certificate and Mac App Store profile
+    script: |
+      app-store-connect fetch-signing-files \
+        "io.codemagic.app" \  # Fetch signing files for specified bundle ID (use "$(xcode-project detect-bundle-id)" if not specified)
+        --platform MAC_OS \
+        --type MAC_APP_STORE \  # Specify provisioning profile type *
+        --create  # Allow creating resources if existing are not found. You may omit this flag if you already have the required certificate and profile and provided the corresponding private key
+  - name: Fetch Mac Installer Distribution certificates
+    script: |
+      app-store-connect create-certificate --type MAC_INSTALLER_DISTRIBUTION --save || \   # You may omit this command if you already have the installer certificate and provided the corresponding private key
+        app-store-connect list-certificates --type MAC_INSTALLER_DISTRIBUTION --save
+  - name: Set up signing certificate
+    script: keychain add-certificates
+  - name: Set up code signing settings on Xcode project
+    script: xcode-project use-profiles
+  ... your build commands
+```
+
+\* Based on the specified bundle ID and [provisioning profile type](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/app-store-connect/fetch-signing-files.md#--typeios_app_adhoc--ios_app_development--ios_app_inhouse--ios_app_store--mac_app_development--mac_app_direct--mac_app_store--mac_catalyst_app_development--mac_catalyst_app_direct--mac_catalyst_app_store--tvos_app_adhoc--tvos_app_development--tvos_app_inhouse--tvos_app_store), Codemagic will fetch or create the relevant provisioning profile and certificate to code sign the build.
```

Comment:
Wow, is that URL really 100 km long? If so ok, if not maybe it can be shortened?

Reply:
![image](https://user-images.githubusercontent.com/18336802/116877994-5eda5100-ac27-11eb-8c66-7a9932be0ea1.png)

to be able to link to a specific argument, they are done as headers. github's markdown doesn't support anchor's aliases, but it's possible to create empty links above so that it will behave the same.

```markdown
<a name="my_super_short_header"></a>
# my super long header
```

but i guess it's fine to do separately for all the tools and arguments

Reply:
no, can't find any normal way to shorten it in github's markdown